commerce-service/src/main/java/com/liferay/commerce/internal/order/term/contributor/CommerceRecipientDefinitionTermContributor.java

```diff
@@ -0,0 +1,247 @@
+			commerceOrder = (CommerceOrder)object;
+		}
+		else if (object instanceof CommerceSubscriptionEntry) {
+			CommerceSubscriptionEntry commerceSubscriptionEntry =
+				(CommerceSubscriptionEntry)object;
+
+			CommerceOrderItem commerceOrderItem =
+				_commerceOrderItemLocalService.getCommerceOrderItem(
+					commerceSubscriptionEntry.getCommerceOrderItemId());
+
+			commerceOrder = commerceOrderItem.getCommerceOrder();
+		}
+
+		if (commerceOrder == null) {
+			return term;
+		}
+
+		if (term.equals(_ORDER_CREATOR)) {
+			User user = _userLocalService.getUser(commerceOrder.getUserId());
+
+			return String.valueOf(user.getUserId());
+		}
+
+		if (term.equals(_ACCOUNT_ROLE_ADMINISTRATOR)) {
+			CommerceAccount commerceAccount =
+				commerceOrder.getCommerceAccount();
+
+			Role accountAdminRole = _roleLocalService.getRole(
+				commerceOrder.getCompanyId(),
+				CommerceAccountConstants.ACCOUNT_ADMINISTRATOR_ROLE_NAME);
+
+			return _getUserIds(commerceAccount, accountAdminRole);
+		}
+
+		if (term.equals(_ACCOUNT_ROLE_ORDER_MANAGER)) {
+			CommerceAccount commerceAccount =
+				commerceOrder.getCommerceAccount();
+
+			Role accountAdminRole = _roleLocalService.getRole(
+				commerceOrder.getCompanyId(), "Order Manager");
+
+			return _getUserIds(commerceAccount, accountAdminRole);
+		}
+
+		if (term.startsWith("%USER_GROUP_")) {
+			String[] s = term.split("_");
+
+			String userGroupName = StringUtil.removeChar(s[2], '%');
+
+			UserGroup userGroup = _userGroupLocalService.getUserGroup(
+				commerceOrder.getCompanyId(), userGroupName);
+
+			return _getUserIds(userGroup);
+		}
+
+		return term;
+	}
+
+	@Override
+	public String getLabel(String term, Locale locale) {
+		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+			"content.Language", locale, getClass());
+
+		return LanguageUtil.get(
+			resourceBundle, _commerceOrderDefinitionTermsMap.get(term));
+	}
+
+	@Override
+	public List<String> getTerms() {
+		return new ArrayList<>(_commerceOrderDefinitionTermsMap.keySet());
+	}
+
+	private String _getUserIds(
+			CommerceAccount commerceAccount, Role accountAdminRole)
+		throws PortalException {
+
+		List<CommerceAccountUserRel> commerceAccountUserRels =
+			_commerceAccountUserRelLocalService.getCommerceAccountUserRels(
+				commerceAccount.getCommerceAccountId());
+
+		StringBundler resultsSB = new StringBundler();
+
+		for (CommerceAccountUserRel commerceAccountUserRel :
+				commerceAccountUserRels) {
+
+			User user = commerceAccountUserRel.getUser();
+
+			List<Role> userRoles = _roleLocalService.getUserRoles(
+				user.getUserId());
+
+			if (userRoles.contains(accountAdminRole)) {
+				resultsSB.append(user.getUserId());
+				resultsSB.append(",");
+			}
+		}
+
+		return resultsSB.toString();
+	}
+
+	private String _getUserIds(UserGroup userGroup) throws PortalException {
+		List<User> groupUsers = _userLocalService.getGroupUsers(
+			userGroup.getGroupId());
+
+		StringBundler results = new StringBundler();
```

Comment:
`StringBundler results` should be named `StringBundler resultsSB`.

And you should be able to initialize the `StringBundler` with a length here.